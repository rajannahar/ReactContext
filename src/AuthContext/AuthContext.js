import React from 'react'

export const AuthContext = React.createContext(false);

const AuthProvider = ({ children }) => {
  const [isLoggedIn, setLoggedIn] = React.useState(false);

  const login = () => {
    setLoggedIn(true)
  }

  const logout = () => {
    setLoggedIn(false)
  }

  return (
    <AuthContext.Provider value={{ login, logout, isLoggedIn }}>
      {children}
    </AuthContext.Provider>
  )
}

export default AuthProvider;