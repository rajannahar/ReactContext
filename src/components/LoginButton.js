import React, {useContext} from 'react'
import { AuthContext } from '../AuthContext/AuthContext'

const LoginButton = () => {
  const { login } = useContext(AuthContext)
  return(
    <button data-testid="loginButton" onClick={ login }>Login</button>
  )
}

export default LoginButton