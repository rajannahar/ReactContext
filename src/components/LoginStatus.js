import React, {useContext} from 'react'
import { AuthContext } from '../AuthContext/AuthContext'

const LoginStatus = () => {

  const status = useContext(AuthContext)
  const {isLoggedIn} = status

  return(
    <>
      <h2>Login status: {isLoggedIn.toString() === "true" ? 'Logged in' : 'Logged out'}</h2>
    </>
  )
}

export default LoginStatus
