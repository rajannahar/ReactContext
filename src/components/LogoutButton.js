import React, {useContext} from 'react'
import { AuthContext } from '../AuthContext/AuthContext'

const LogoutButton = () => {
  const { logout } = useContext(AuthContext)
  return(
    <button data-testid="logoutButton"  onClick={() => logout()}>Logout</button>
  )
}

export default LogoutButton;