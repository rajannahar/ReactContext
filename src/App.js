import React from 'react';
import { AuthProvider } from './AuthContext/AuthContext'

import LoginStatus from './components/LoginStatus'
import LoginButton from './components/LoginButton'
import LogoutButton from './components/LogoutButton'

function App() {
  return (
    <div className="App">
      <h1>React Context</h1>
      <AuthProvider>
        <LoginStatus />
        <LoginButton />
        <br />
        <LogoutButton />
      </AuthProvider>
    </div>
  );
}

export default App;