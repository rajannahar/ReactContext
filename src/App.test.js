import React, { useContext } from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react'
import { getByTestId, queryByTestId, getByText, waitForElement } from '@testing-library/dom'
import { AuthContext, AuthProvider } from './AuthContext/AuthContext';
import LoginButton from './components/LoginButton'
import LogoutButton from './components/LogoutButton'
import LoginStatus from './components/LoginStatus'

import App from './App'

afterEach(cleanup)

test('should render login button', () => {
  const { getByTestId } = render(<App />)
  const loginButton = getByTestId('loginButton')
  expect(loginButton).toBeTruthy()
})

test('should render logout button', () => {
  const { getByTestId } = render(<App />)
  const logoutButton = getByTestId('logoutButton')
  expect(logoutButton).toBeTruthy()
})

test('should render login status', () => {
  const { getByTestId } = render(<App />)
  const heading = getByTestId('loginStatus')
  expect(heading.innerHTML).toBe('Login status: Logged out')
})

test('click login button should update text', async () => {
  const { getByTestId, getByText } = render(<App />)

  const headingInitial = getByTestId('loginStatus')
  const loginButton = getByText('Login')
  expect(headingInitial.innerHTML).toEqual('Login status: Logged out')

  fireEvent.click(loginButton)

  const headingLogin = await waitForElement(() => getByTestId('loginStatus'))
  expect(headingLogin.innerHTML).toEqual('Login status: Logged in')
})

it("matches snapshot", () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});