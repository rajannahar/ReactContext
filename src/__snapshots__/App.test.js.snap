// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
    data-testid="appTest"
  >
    <h1>
      React Context
    </h1>
    <h2
      data-testid="loginStatus"
    >
      Login status: Logged out
    </h2>
    <button
      data-testid="loginButton"
    >
      Login
    </button>
    <br />
    <button
      data-testid="logoutButton"
    >
      Logout
    </button>
  </div>
</DocumentFragment>
`;
